### PR TITLE
Dashboard implementation

### PR DESCRIPTION
**Pull request by WRIOLtd**
https://github.com/webRunes/WRIO-InternetOS/issues/1581

### CODE REVIEW

File: src/base/components/Dashboard.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      sensorData: []
+    }
+  }
+
+  componentDidMount() {
+    setTimeout(() => {
```
Comment (AlexeyAnshakov) [inline src/base/components/Dashboard.js]: what this? we can't rely on load timing.

File: src/base/components/Dashboard.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      this.setState({ sensorData: this.props.sensorData })
+    }, 3000);
+    document.getElementsByClassName('main')[0].style.width = "1300px";
```
Comment (AlexeyAnshakov) [inline src/base/components/Dashboard.js]: No styles in the code

File: src/base/components/Dashboard.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            <th>Location </th>
+            <th>Access </th>
+            <th>Cost </th>
+            <th>Protocol </th>
+            <th>Role </th>
+            <th>Alerts </th>
+            <th>State </th>
+            <th>Last seen</th>
+            <th>Last measurement</th>
+            <th>Device battery measurement</th>
```
Comment (AlexeyAnshakov) [inline src/base/components/Dashboard.js]: Delete "measurement".

**Review by AlexeyAnshakov — CHANGES_REQUESTED** [review]

File: src/base/components/Dashboard.js (hunk 1 above)
Comment (WRIOLtd) [reply src/base/components/Dashboard.js]: Sorry forgot to remove

File: src/base/components/Dashboard.js (hunk 2 above)
Comment (WRIOLtd) [reply src/base/components/Dashboard.js]: But the "main div" width is quite small and I have to increase it to adjust all table columns!

File: src/base/components/Dashboard.js (hunk 3 above)
Comment (WRIOLtd) [reply src/base/components/Dashboard.js]: Device battery measurement to Device battery ?

File: src/base/components/Dashboard.js (hunk 1 above)
Comment (WRIOdev) [reply src/base/components/Dashboard.js]: not using this

File: src/base/components/Dashboard.js (hunk 2 above)
Comment (AlexeyAnshakov) [reply src/base/components/Dashboard.js]: You can add style into .css next time.
But there no need. We will update the table like i said before. Currently we need only working code.

File: src/base/components/Dashboard.js (hunk 3 above)
Comment (AlexeyAnshakov) [reply src/base/components/Dashboard.js]: See https://github.com/webRunes/WRIO-InternetOS/issues/1581#issuecomment-499443413
All must be as described

File: src/base/components/Dashboard.js (hunk 1 above)
Comment (WRIOLtd) [reply src/base/components/Dashboard.js]: removed!

File: src/base/components/Dashboard.js (hunk 2 above)
Comment (WRIOLtd) [reply src/base/components/Dashboard.js]: my code editor unable to find any .css file ?
are css files added through hosted links ?

Comment (AlexeyAnshakov) [reply src/base/components/Dashboard.js]: https://github.com/webRunes/Default-WRIO-Theme
But don't do any changes at the moment. Just remove styles in the code.

File: src/base/components/Dashboard.js (hunk 3 above)
Comment (WRIOLtd) [reply src/base/components/Dashboard.js]: updated

File: src/base/components/Dashboard.js (hunk 2 above)
Comment (WRIOLtd) [reply src/base/components/Dashboard.js]: Ok

Comment (WRIOLtd) [reply src/base/components/Dashboard.js]: removed!

**Comment by AlexeyAnshakov** [comment]
Please add a link to an issue's description next time: see above, added as an example.

**Pull request merged**